lib/docker-toolbelt.coffee

```diff
@@ -302,6 +302,38 @@ DockerToolbelt::createEmptyImage = (imageConfig) ->
 				stream.pipe(es.wait(callback))
 		.return(imageId)
 
+# Given a source and destination image, generates a delta and returns a promise
+# that resolves with the ID of the generated image. `onProgress` is an optional
+# callback that receives a single argument for the progress event that can used
+# to follow progress.
+#
+# Deltas are currently only available with Balena, but this method makes no
+# effort to determine whether that's the case.
+#
+# The name of this method is intentionally unconventional for docker-toolbelt,
+# anticipating the appearance of a similar method *in* dockerode in the future.
+DockerToolbelt::createDeltaAsync = (src, dest, onProgress) ->
+	optsf = {
+		path: '/images/delta?',
+		method: 'POST',
+		options: { src, dest },
+		isStream: true,
+		statusCodes:
+			200: true,
+			404: 'no such image',
+			500: 'server error',
+	}
+	return new @modem.Promise (resolve, reject) =>
```

Comment:
You can simplify this by using
```
Promise.fromCallback (cb) =>
	@modem.dial(optsf, cb)
```
since we force the modem to a bluebird promise in the constructor